### ISSUE TITLE
false postive with names starting with a number using jsx-pascal-case

### ISSUE THREAD

**Issue by mfyuce** [yannickcr/eslint-plugin-react#1336]
(Not directly about run. It is about the RegEx provided in the code.)

Using v7.1.0,
```jsx
<1MyComponent/>
<2MyComponent/>
<3MyComponent/>
```

do not flag an error (e.g Contained JSX component MyComponent must not start with a number (react/jsx-pascal-case))

**Comment by ljharb** [comment]
This seems fine to me, since those are invalid identifiers, they shouldn't work in the first place.

**Comment by mfyuce** [comment]
True, but the regex, that is checking the IDs, is not covering this case. But somehow, in some place, it gives some error when linting, but not (i think) using methods within this project.

 I can close the issue, if you want.

**Comment by ljharb** [comment]
The jsx parser should be throwing on it because it's invalid JS. The regex for this rule shouldn't need to check it, because it should be impossible for it to receive that in the first place.

**Comment by mfyuce** [comment]
That is reasonable. Thank you.
